evalml/model_understanding/prediction_explanations/_user_interface.py

```diff
@@ -423,3 +451,25 @@ def make_dict(self, data):
                                                                  data.input_features)["explanations"]
             report.append(section)
         return {"explanations": report}
+
+    def make_dataframe(self, data):
+        report = []
+        for rank, index in enumerate(data.index_list):
+            shap_table = self.table_maker.make_dataframe(index, data.pipeline, data.input_features)
+            shap_table['rank'] = rank
+            if self.make_predicted_values_maker:
+                heading = self.make_predicted_values_maker.make_dataframe(index, data.y_pred, data.y_true, data.errors,
+                                                                          pd.Series(data.input_features.index))
+                for key, value in heading.items():
+                    if key == "probabilities":
+                        for class_name, probability in value.items():
+                            shap_table[f'label_{class_name}_probability'] = probability
+                    else:
+                        shap_table[key] = value
+            if self.heading_maker:
+                heading = self.heading_maker.make_dataframe(rank)
+                shap_table['rank'] = heading['index']
+                shap_table['prefix'] = heading['prefix']
+            report.append(shap_table)
+        df = pd.concat(report).reset_index(drop=True)
```

Comment:
So this code will create one df for each prediction, and then concatenate all the columns together horizontally? How do users know which columns correspond to which prediction?

Reply:
To resolve: They'll use the `prediction_number` column we talked about above!